### PR TITLE
[WIP] OptimizeHopper: update optimized solution plots

### PR DESCRIPTION
**Pull request by nickbianco**
### Brief summary of changes
I updated the solution plots so they're more informative and interesting. 

### Looking for feedback on...
The implementation is as concise as I could make it on a first pass, but it's not very clean. I put the plotting in a new subfunction `plotSolution()` to address this, but feedback would be helpful.

### CODE REVIEW

File: Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m [hunk 1]
```diff
@@ -55,35 +55,15 @@
         lowerBound = 0.25*ones(numVars, 1); % seconds
         upperBound = 0.75*ones(numVars, 1); % seconds
         options = optimoptions('ga','FunctionTolerance', 1e-2, ...
-                                    'MaxGenerations', 10, ...
+                                    'MaxGenerations', 1, ...
```
Comment (chrisdembia) [inline Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m]: You can still get 3 nice hops with 1 generation?

File: Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m [hunk 2]
```diff
@@ -201,4 +182,57 @@
...
+subplot(2,1,1);
+plot(deviceControl(1,:), deviceControl(2,:), 'b-', 'linewidth', 2)
+hold on
+plot(muscleExcitation(1,:), muscleExcitation(2,:), 'r-', 'linewidth', 2)
+ylim([-0.10 1.10]); xlim([0 5]); xticks(0:1:5);
+switch problem
+    case 'one_hop'
+        text_x = [x/2 t];
+        text_y = [1.05 -0.05];
+        labels = {'<--x(1)-->', 't1'};
```
Comment (chrisdembia) [inline Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m]: I wonder if the `<--x(1)-->` will consistently fill up the correct amount of space on different computers or for different figure sizes; perhaps there are cases where the `<...>` extend beyond the desired interval.

File: Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m [hunk 3]
```diff
@@ -201,4 +182,57 @@
...
+    case 'three_hops'
+        text_x = [(x([1 2 1 3 1])/2 + [0 t([1 2 3 4])]) t];
+        text_y = [1.05*ones(1,5) -0.05*ones(1,5)];
```
Comment (chrisdembia) [inline Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m]: Nice!

File: Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m [hunk 4]
```diff
@@ -201,4 +182,57 @@
...
+        x_labels = arrayfun(@(x) sprintf('<--x(%i)-->', x), [1 2 1 3 1], ...
+            'UniformOutput', false);
```
Comment (chrisdembia) [inline Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m]: I did not know about `arrayfun()` but you should be able to get "iteration" using `sprintf()` on its own. I believe that `sprintf('%i ', [1, 2, 3])` will print `1 2 3` (it should repeat the string pattern).

Comment (nickbianco) [reply Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m]: `sprintf()` will do the iteration, but returns a string. I need a cell array of strings for the `text()` argument, which `arrayfun(@func, 'UniformOutput', false)` returns. 

The other option to get a cell array would be `strsplit(sprintf('<--x(%i)-->,', [1 2 1 3 1]), ',')`. It gives an extra empty string in the last cell, but that could be ignored when creating the final label array:
`labels = [x_labels(1:end-1) t_labels(1:end-1)]`. This method would also avoid the line continuation that `arrayfun()` needs for the `UniformOutput` argument.

File: Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m (hunk 1 above)
Comment (nickbianco) [reply Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m]: I ran it a few times with one generation, and I'm still getting solid triple jumps (close to 1.75 if not above). So we could probably get away with one generation.

**Comment by chrisdembia** [comment]
I ran this locally and the plots look nice. I like the ⭐️ at the peak.

**Comment by chrisdembia** [comment]
You've marked the PR as `[WIP]` so I won't merge it; feel free to merge it yourself when you're ready.

**Comment by chrisdembia** [comment]
@nickbianco do you plan on adding more to this PR?

**Pull request merged**